### PR TITLE
Typescript camera console

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
Refactors the camera console into typescript, including the secureye, removing the need for its custom css file

KaBYZCTqG1

Screenshot 2023-09-17 120550
## Why It's Good For The Game
Spruces up the UI ever so slightly and possibly fixes an issue #78009

Less code is better code
## Changelog
:cl:
refactor: Refactors the camera console UI.
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/CameraConsole.tsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+ * Camera selector.
+ *
+ * Filters cameras, applies search terms and sorts the alphabetically.
+ */
+const selectCameras = (cameras: Camera[], searchText = ''): Camera[] => {
+  const testSearch = createSearch(searchText, (camera: Camera) => camera.name);
+
+  return flow([
+    // Null camera filter
+    filter((camera: Camera) => !!camera?.name),
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/CameraConsole.tsx:48]: it's possible that the issue was here. without digging into the backend and just reworking how this works entirely (null camera names causes another issue), typescript was complaining that this needed to be casted into a boolean

**Review by stylemistake — APPROVED** [review]

**Comment by SyncIt21** [comment]
Please link issue properly in the PR body I.e.

Fixes #78009

Not

Fixes an issue #78009

Great work

**Comment by jlsnow301** [comment]
> Fixes an issue #78009

there was a reason I did this, I cant reproduce their error and I'm not certain this would fix it

**Comment by jlsnow301** [comment]
> rarely is a bsod reported with reproduction steps that are valid

not their fault but i can't even see what the error entails in the original issue, just that there is one 😔 we could close the original i guess until the problem reappears

**Pull request merged**